custom/abt/reports/filters.py

```diff
@@ -1,122 +1,103 @@
 from django.utils.translation import ugettext_lazy
-from sqlagg.columns import SimpleColumn
-from sqlagg.sorting import OrderBy
 
 from corehq.apps.reports.filters.base import BaseSingleOptionFilter
-from corehq.apps.reports.sqlreport import SqlData, DatabaseColumn
-from corehq.apps.userreports.util import get_table_name
-from corehq.sql_db.connections import DEFAULT_ENGINE_ID
-
-
-class FilterDataSource(SqlData):
-    engine_id = DEFAULT_ENGINE_ID
-
-    def __init__(self, domain, loc_level):
-        self.config = {}
-        self.domain = domain
-        self.loc_level = loc_level
-
-    @property
-    def table_name(self):
-        return get_table_name(self.domain, "static-late-pmt")
-
-    @property
-    def filters(self):
-        filters = []
-        return filters
-
-    @property
-    def group_by(self):
-        return [self.loc_level]
-
-    @property
-    def order_by(self):
-        return [OrderBy(self.loc_level)]
-
-    @property
-    def columns(self):
-        return [
-            DatabaseColumn(self.loc_level, SimpleColumn(self.loc_level))
-        ]
-
-
-class UserFilterDataSource(SqlData):
-    engine_id = DEFAULT_ENGINE_ID
-
-    def __init__(self, domain):
-        self.config = {}
-        self.domain = domain
-
-    @property
-    def table_name(self):
-        return get_table_name(self.domain, "static-late-pmt")
-
-    @property
-    def filters(self):
-        filters = []
-        return filters
-
-    @property
-    def group_by(self):
-        return ['doc_id', 'username']
-
-    @property
-    def order_by(self):
-        return [OrderBy('username')]
-
-    @property
-    def columns(self):
-        return [
-            DatabaseColumn('doc_id', SimpleColumn('doc_id')),
-            DatabaseColumn('username', SimpleColumn('username'))
-        ]
+from custom.abt.reports.fixture_utils import (
+    get_data_types_by_tag,
+    get_fixture_dicts,
+)
 
 
 class VectorLinkLocFilter(BaseSingleOptionFilter):
     default_text = 'All'
 
-    @property
-    def options(self):
-        data = FilterDataSource(self.domain, self.slug).get_data()
-        options = [(x[self.slug], x[self.slug]) for x in data]
-        return options
-
-
-class UsernameFilter(VectorLinkLocFilter):
-    slug = 'user_id'
-    label = ugettext_lazy('Username')
-
-    @property
-    def options(self):
-        data = UserFilterDataSource(self.domain).get_data()
-        options = [(x['doc_id'], x['username']) for x in data]
-        return options
-
+    def get_level_2s(self, level_1_ids):
+        data_types_by_tag = get_data_types_by_tag(self.domain)
+        return get_fixture_dicts(
+            self.domain,
+            data_types_by_tag["level_2_dcv"]._id,
+            filter_in={'level_1_dcv': level_1_ids}
+        )
 
-class CountryFilter(VectorLinkLocFilter):
```

Comment:
Is it one country per domain?

Reply:
It seems so, although I don't know if that is a hard rule. I'll confirm. [All 26 domains](https://github.com/dimagi/commcare-hq/blob/bbb5a7d841890b743c9bb970f54145ed0283e523/settings.py#L2012) include the name of a country, except for "airs" and "airs-testing". "master-vectorlink" is the linked-app master, but doesn't have the custom reports.